### PR TITLE
Added TypeScript defintions

### PR DESCRIPTION
**Pull request by chr33s**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,7 +45,7 @@
         "@babel/preset-env",
         {
           "targets": {
-            "node": "12"
+            "node": "10"
```
Comment (vvo) [inline package.json:48]: hey there, why change node.js target to 10 instead of 12?

File: .gitignore [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 # misc
 .DS_Store
-/dist/
+/dist/*.map
```
Comment (vvo) [inline .gitignore:7]: Why do we need to publish the library build to git? I'd prefer to keep `/dist/` here and only have the build inside the npm package.

Comment (chr33s) [reply .gitignore]: removed, published by accident, was a hot fix of sorts.

File: package.json (hunk 1 above)
Comment (chr33s) [reply package.json]: removed, was a debug artefact.

File: package.json [hunk 3]
```diff
@@ -20,16 +20,19 @@
   "sideEffects": false,
   "main": "dist/",
   "module": "lib/",
+  "types": "index.d.ts",
```
Comment (vvo) [inline package.json:23]: should this be dist/index.d.ts or index.d.ts? I am asking because in postbuild we move index.d.ts into dist. thanks!

Comment (chr33s) [reply package.json]: Sorry, corrected.

**Comment by vvo** [comment]
Thanks! This will be released soon.

**Pull request merged**